website/static/js/addonNodeConfig.js

```diff
@@ -0,0 +1,179 @@
+        });
+        
+        // Overrides
+        self.options = {
+            onPickFolder: function(evt, item) {
+                evt.preventDefault();
+                var name = item.data.path !== '/' ? item.data.path : '/ (Full ' + self.addonName + ')';
+                self.selected({
+                    name: name,
+                    path: item.data.path,
+                    id: item.data.id
+                });
+                return false; // Prevent event propagation
+            }
+        };
+        // Overrides
+        self.options = $.extend(self.options, opts);
+        // Treebeard config
+        self.treebeardOptions = $.extend(
+            {}, 
+            FolderPickerViewModel.prototype.treebeardOptions,
+            {
+                onPickFolder: function(evt, item) {
+                    return this.options.onPickFolder.call(this, evt, item);
+                }.bind(this),
+                resolveLazyloadUrl: function(item) {
+                    return item.data.urls.folders;
+                }
+            }
+        );
+    },
+    toggleShare: function() {
+        if (this.currentDisplay() === this.SHARE) {
+            this.currentDisplay(null);
+        } else {
+            // Clear selection
+            this.cancelSelection();
+            this.currentDisplay(this.SHARE);
+            this.activateShare();
+        }
+    },
+    fetchEmailList: function(){
+        var self = this;
+        
+        var ret = $.Deferred();
+        var promise = ret.promise();
+        if(!self.loadedEmails()){
+            $.ajax({
+                url: self.urls().emails,
+                type: 'GET',
+                dataType: 'json'
+            }).done(function(res){
+                self.loadedEmails(true);
+                ret.resolve(res.results.emails);
+            }).fail(function(xhr, status, error){
+                Raven.captureMessage('Could not GET ' + self.addonName + ' email list', {
+                    url: self.urls().emails,
+                    textStatus: status,
+                    error: error
+                });
+                ret.reject(xhr, status, error);
+            });
+        }
+        else{
+            ret.resolve(self.emails());
+        }
+        return promise;
+    },
+    activateShare: function() {
+        var self = this;
+        self.fetchEmailList()
+            .done(function(emails){
+                self.emails(emails);
+            });
+        var $copyBtn = $(self.selector).find('.copyBtn');
+        new ZeroClipboard($copyBtn);
+    },
+    _updateCustomFields: function(settings){
+        this.validCredentials(settings.validCredentials);
+        this.canShare(settings.canShare || false);
+    },
+    fetchAccounts: function() {
+        var self = this;
+        var request = $.get(self.urls().accounts);
+        request.then(function(data) {
+            return data.accounts;
```

Comment:
@sloria  My intention is to unpack the response before it gets passed to the ```done()``` callback in ```updateAccounts```.